src/core.js

```diff
@@ -64,29 +64,29 @@ function createModule( name, testEnvironment ) {
 	return module;
 }
 
-function processModule( name, testEnvironment, executeNow ) {
-	let module = createModule( name, testEnvironment );
+function processModule( name, options, executeNow ) {
+	let module = createModule( name, options );
 
 	// Move any hooks to a 'hooks' object
-	if ( module.testEnvironment ) {
-		module.hooks = {
-			before: module.testEnvironment.before,
-			beforeEach: module.testEnvironment.beforeEach,
-			afterEach: module.testEnvironment.afterEach,
-			after: module.testEnvironment.after
-		};
-
-		delete module.testEnvironment.before;
-		delete module.testEnvironment.beforeEach;
-		delete module.testEnvironment.afterEach;
-		delete module.testEnvironment.after;
+	const testEnvironment = module.testEnvironment;
+	const hooks = module.hooks = {};
+
+	setHookFromEnvironment( hooks, testEnvironment, "before" );
+	setHookFromEnvironment( hooks, testEnvironment, "beforeEach" );
+	setHookFromEnvironment( hooks, testEnvironment, "afterEach" );
+	setHookFromEnvironment( hooks, testEnvironment, "after" );
+
+	function setHookFromEnvironment( hooks, environment, name ) {
```

Comment:
Unsure what you mean as both parameters are used in the function. Unless you're referring to the function itself, which is really just here to avoid repeating the same code for each hook.